data-prepper-core/src/main/java/com/amazon/dataprepper/pipeline/server/DataPrepperServer.java

```diff
@@ -48,4 +65,42 @@ public void stop() {
         server.stop(0);
         LOG.info("Data Prepper server stopped");
     }
+
+    private HttpServer createHttpServer(final int port) throws IOException {
+        return HttpServer.create(new InetSocketAddress(port), 0);
+    }
+
+    private HttpServer createHttpsServer(final int port,
+                                         final String keystoreFilePath,
+                                         final String passphrase) throws IOException {
+        final SSLContext sslContext;
+
+        try {
+            char[] passphraseArray = passphrase.toCharArray();
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(new FileInputStream(keystoreFilePath), passphraseArray);
+
+            KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance("SunX509");
+            keyManagerFactory.init(keyStore, passphraseArray);
+
+            TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance("SunX509");
+            trustManagerFactory.init(keyStore);
+
+            sslContext = SSLContext.getInstance("TLS");
+            sslContext.init(keyManagerFactory.getKeyManagers(), trustManagerFactory.getTrustManagers(), null);
+        } catch (Exception e) {
+            throw new IllegalStateException("Problem loading keystore to enable TLS", e);
```

Comment:
Question: Should we add a LOG.error here ? Not an issue as the calling code is only catching `IOException` but thinking, how the logs will be. I am fine either ways

Reply:
The unhandled exception will bubble up in the logs, don't think we need to double log